src/main/java/com/minecolonies/coremod/commands/generalcommands/ScanCommand.java

```diff
@@ -0,0 +1,388 @@
+        }
+
+        final String[] split = fileName.split("/");
+        final String style = split.length <= 1 ? "" : split[0];
+
+        final BlockPos zero = new BlockPos(box.minX(), box.minY(), box.minZ());
+        final Blueprint
+          bp = BlueprintUtil.createBlueprint(world, zero, saveEntities, (short) box.getXSpan(), (short) box.getYSpan(), (short) box.getZSpan(), fileName, slot.getBox().getAnchor());
+
+        if (slot.getBox().getAnchor().isEmpty() && bp.getPrimaryBlockOffset().equals(new BlockPos(bp.getSizeX() / 2, 0, bp.getSizeZ() / 2)))
+        {
+            final List<BlockInfo> list = bp.getBlockInfoAsList().stream()
+                                           .filter(blockInfo -> blockInfo.hasTileEntityData() && blockInfo.getTileEntityData().contains(TAG_BLUEPRINTDATA))
+                                           .collect(Collectors.toList());
+
+            if (list.size() > 1)
+            {
+                player.displayClientMessage(Component.translatable("com.ldtteam.structurize.gui.scantool.scanbadanchor", fileName), false);
+            }
+        }
+
+        Network.getNetwork().sendToPlayer(new SaveScanMessage(BlueprintUtil.writeBlueprintToNBT(bp), fileName), (ServerPlayer) player);
+        if (style.isEmpty())
+        {
+            return;
+        }
+
+        boolean hasJigsaw = false;
+        for (BlockState state : bp.getPalette())
+        {
+            if (state.getBlock() == Blocks.JIGSAW)
+            {
+                hasJigsaw = true;
+                break;
+            }
+        }
+
+        if (!hasJigsaw)
+        {
+            return;
+        }
+
+        final BlockState primary = world.getBlockState(zero.offset(bp.getPrimaryBlockOffset()));
+        final boolean isHut;
+        if (primary.getBlock() instanceof AbstractBlockHut<?>)
+        {
+            isHut = true;
+            final AbstractTileEntityColonyBuilding building = (AbstractTileEntityColonyBuilding) world.getBlockEntity(zero.offset(bp.getPrimaryBlockOffset()));
+            building.addTag(new BlockPos(0, 0, 0), "deactivated");
+            building.addTag(new BlockPos(0, 0, 0), style);
+        }
+        else
+        {
+            isHut = false;
+        }
+
+        for (final BlockPos mutablePos : BlockPos.betweenClosed(zero, zero.offset(box.getXSpan() - 1, box.getYSpan() - 1, box.getZSpan() - 1)))
+        {
+            BlockState state = world.getBlockState(mutablePos);
+            if (state.getBlock() == Blocks.JIGSAW)
+            {
+                JigsawBlockEntity jigsawBlockEntity = (JigsawBlockEntity) world.getBlockEntity(mutablePos);
+                if (jigsawBlockEntity.getName().getPath().isEmpty() || jigsawBlockEntity.getName().getPath().equals("empty"))
+                {
+                    if (jigsawBlockEntity.getFinalState().equals("minecraft:air"))
+                    {
+                        jigsawBlockEntity.setFinalState("minecraft:structure_void");
+                    }
```

Comment:
`structure_void` is equivalent to the light placeholder, not to air.  This means that this is possibly going to leave a solid block in that location when that was probably not intended.  Why is this not just leaving it as air?

People can set it to `structure_void` themselves if they want it to be a placeholder.  (Or, presumably, to a light placeholder, although that might not work depending on what order it runs the processors.)

Reply:
all the current ones in the mod use structure void at the place, thats why I went with this